Update release documentation for 3.3.0 (#25401)
wxWidgets/wxWidgets: docs/msw/binaries.md

Supported Compilers
-------------------

We provide pre-built binary files for x86-64 Windows for the following compilers:

* Microsoft Visual C++ compiler versions 14.0, 14.1, 14.2 and 14.3
  (corresponding to marketing product names of Microsoft Visual Studio 2015, 2017, 2019 and 2022 respectively).
  Please note that MSVC 14.x versions are ABI-compatible and the same set of binaries is used for all of them.

> * [MSYS2](https://www.msys2.org/) MinGW 15.1.

Reviewer: ```suggestion
* [MSYS2](https://www.msys2.org/) environment MINGW64 (GCC 15.1).
```
Just to use the official MSYS2 environment name. MinGW there can also use clang, I think. I guess it was decided to stick to the old MSVCRT package instead of the newer one using UCRT (i.e., MINGW64 over UCRT64) for compatibility (no need to distribute UCRT on Windows < 10). See https://www.msys2.org/docs/environments/

Getting the files
-----------------

First, you need to get the correct files. You will always need the
`wxWidgets-3.3.0-headers.7z` one but the "Dev" and the "ReleaseDLL"
files depend on your compiler: for example, for MSVS you will need
`wxMSW-3.3.0_vc14x_x64_Dev.7z` and `wxMSW-3.3.0_vc14x_x64_ReleaseDLL.7z`, but

> if you are using MinGW-w64, you will need

Reviewer: ```suggestion
if you are using MINGW64, you will need
```

Reviewer: I used "MINGW64" everywhere as it is the official [MSYS2 environment](https://www.msys2.org/docs/environments/) name.

I believe (perhaps incorrectly), that here (identifying an MSYS2 compiler toolchain), _mingw-w64_ means something different that _MINGW64_. mingw-w64 provides headers and some libraries and tools (and no longer the full compiler toolchain). What mingw-w64 provides is used in different MSYS2 environments: Different compilers (GCC vs clang), architectures (e.g., i686 vs x86_64), or even the same compiler and architecture (GCC and x86_64) but different C runtime (MINGW64 vs UCRT64).

IMO, the official MSYS2 environment name MINGW64 clearly and unequivocally identifies the one platform the binaries are for: GCC compiler, x64_86 architecture, and using MSVCRT (as opposed to UCRT64 using UCRT)

Author: And here I think just using `gcc` should be clear enough.

> `wxMSW-3.3.0_gcc810_x64_Dev.7z` and `wxMSW-3.3.0_gcc810_x64_ReleaseDLL.7z`.

Reviewer: ```suggestion
`wxMSW-3.3.0_gcc1510_x64_Dev.7z` and `wxMSW-3.3.0_gcc1510_x64_ReleaseDLL.7z`.
```

Author: Fixed, thanks.
